events/schema.md

```diff
@@ -4,30 +4,159 @@ Events published to the bus have a standard schema that allows SDKs and services
+            "identifier": "6551f09a-bd23-4f12-a039-f8fa78ef776a",
+        },
+        {
+            "service": "downstream-sample",
+            "name": "deposit",
+            "identifier": "2018-03-08/1",
+        },
+    ],
 }
 ```
 
-- `eventId` is unique for all events. It is a UUID version 1 generated with time and a node identifier.
-- `happenedAt` identifies when the event has happened (then it can get published, consumed, etc).
-- `aggregate` identifies the unit of consistency that has produced the event: article run in the bot, article version in the articles, store podcast episode in journal-cms, an article's set of metrics in metrics, a single profile in profiles. All three information are required to uniquely identify it: `service`, `name`, `identifier`.
-- `type` describes what the event is about so that they can be grouped or recognized.
-- `data` is opaque here, and can contain anything as long as it follows naming conventions. Events with the same `type` should usually follow the same schema.
-- `correlationId` can be used to track events across services, grouping them by the kind of session that originated them. Could be an article run during ingestion, or a journal page creating a profile. It is optional at this time.
+## Fields
+
+### `eventId`
+
+Unique for all events. It is a UUID version 1 generated with time and a node identifier.
+
+### `happenedAt` 
+
+Identifies when the event has happened (then it can get published, consumed, etc). It follows the `yyyy-mm-ddTHH:MM:SS+00:00` ISO 8601 format, and is expressed in UTC.
+
+### `entity`
+
+Identifies the unit of consistency that has produced the event: article run in the bot, article version in the articles store, podcast episode in journal-cms, an article's set of metrics in metrics, a single profile in profiles.
+
+All three information are required to uniquely identify it:
+
+- `service`: `[a-z\-]+`
+- `name`: `[a-z\-]+`
+- `identifier`: `.+` (possibly URL-friendly)
+
+This identifier may or may not correspond to a REST resource accessible through the API of the originating service.
+
+### `type`
+
+Describes what the event is about so that they can be grouped or recognized. It follows the `[a-z\-\.]+` regular expression, with `.` separating hierarchical levels and `-` separating words in a single level (if there are multiple words).
+
+### `data` (optional)
+
+Is opaque here, and can contain anything as long as it follows naming conventions. Events with the same `type` should usually follow the same schema.
+
+### `correlationIds` (optional)
+
+Can be used to track events across services, grouping them by the kind of session that originated them or the data type they refer to. Could be an article run during ingestion, or a journal page creating a profile.
+
+Represents an array of dictionaries, each following the format of `entity`.
+
+Services should normally add correlation ids from events they have triggered them to the new events being produced.
```

Comment:
Looking at https://github.com/libero/walking-skeleton/pull/28#discussion_r190211098, _might_ be a replacement?